### PR TITLE
fix: Messaging v2 application creation endpoints. Add v2 API typescript types.

### PR DESCRIPTION
**Pull request by amiller-gh**
Hi all!

While integrating my account with our new admin tooling for application creation I discovered a number of issues with the Messaging V2 application creation APIs:

 - Base URL appears to be wrong
 - `CallBackCreds` => `CallbackCreds` for application creation.
 - `protocol` => `Protocol` in MMS enable
 -  `POST` => `PUT` in `_assignApplicationToLocation`
 
The one other issue I ran into was when the SDK passes `ProxyPeerId` in `_enableMms` and `_enableSms`. Even with an "admin" user's credentials it always failed with `UnexpectedResponseError: 1001: Only admin users are allowed to set peer id`. All API methods still work when this property is removed. I've left it commented out in this PR, please feel free to chime in there!

I've also taken the liberty of adding type definitions for the entire Messaging V2 SDK. I don't want to be *that* OSS contributor, so please feel free to tell me to nix them from the PR! But, wanted to offer them to you anyway – my code base would certainly appreciate it!

---

Make sure you've checked off all these things before submitting:

 - [x] This pull request contains 100% test coverage.
 - [x] This pull request is completely documented. All available fields are listed in the docs with description.
 - [x] Each public facing function should include at least one @example in the jsdoc.
 - [x] Run `npm test` before committing.

### CODE REVIEW

**Comment by amiller-gh** [comment]
Thread bump 🙏 Would love to be able to get off a personal fork in production

File: docs/api.md [hunk 1]
```diff
@@ -1,3720 +1,4067 @@
...
-		console.log(response.transactions);
-	});
-```
-**Example**  
-```js
-//Get transactions filtering by date
-//Promise
-var params = {
-	fromDate: "2013-02-21T13:38:00"
+var util = require("util");
```
Comment (amiller-gh) [inline docs/api.md:1]: Whoops, I overwrite api.md! Don't merge this file. – the rest is still applicable 👍

Comment (BryanZWu) [reply docs/api.md:1]: Hey @amiller-gh 
I checked the team. Could you check if https://github.com/bandwidth/node-messaging has what you're looking for? It's the newer SDKs, I think.

**Comment by dtolb** [comment]
@amiller-gh thanks x100 for the work here. I'm going to bring on @jmulford-bw to merge and release (will be a major bump due to the previous typos).

Apologies on the delay here, we've been working exclusively on our next-gen SDKs and driving towards a complete TS version  this fall. And completely missed any v1 updates. 

We'll get this in for you sooner than later now :)

**Comment by jmulford-bw** [comment]
Hey @amiller-gh thanks for doing this! Nothing seems of concern to me, but can you update the version in `package.json`? I'm guessing it'll be `4.0.0` if this is a breaking change

**Comment by amiller-gh** [comment]
Ahh thanks for taking that on @jchavez443! Sorry for the delay.

@BryanZWu Love the look of those new bindings, but they appear to be missing the application creation endpoints? We use this package for our automated account creation for clients. Correct me if I'm wrong :)

**Comment by BryanZWu** [comment]
@amiller-gh I think you'd be right about the application creation endpoints. For that, I think https://github.com/Bandwidth/node-numbers has what you're looking for

**Pull request merged**